lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -307,7 +344,7 @@ def read_role_secret(self, name: str, id: str) -> dict:
         response = self._client.auth.approle.read_secret_id(name, id)
         return response["data"]
 
-    def enable_secrets_engine(self, backend_type: SecretsBackend, path: str) -> None:
+    def ensure_secrets_engine(self, backend_type: SecretsBackend, path: str) -> None:
```

Comment:
Why the name change? I'm not convinced `ensure` is better

Reply:
ensure implies (to me) that it is idempotent, whereas enable doesn't. I was trying to address the same issue we had with `configure_*`, and we seemed to be converging on `create_or_update_` for those.

I can change it back if it's disliked.

Reply:
`create_or_update_` won't create or update anything if the field is already in its correct value right? Ensure is too vague in my opinion, ensure what? That it is enabled, disabled, created, etc. I do agree though on having every method be idepotent, but I don't think it should mean renaming all of them to `ensure_xyz`

Reply:
https://github.com/canonical/vault-k8s-operator/pull/534/commits/53f215012786fc13fb764915c729fee1a0c047ef